python/modules/variable.py

```diff
@@ -89,53 +89,122 @@ def get_value(self, env, args = {}):
+
+        print('Python version does not meet software requirments. '
+        'Install python 3.5 or 3.6 to run IPIQA.')
+
+    def get_value(self, env):
+        args_keys = (self.args).keys()
+        if "well" in args_keys:
+            well = self.args["well"].get_value(env)
+            try:
+                self.mp_dict = env[self.value].get_mp_dict()
+            except:
+                logger.error("Following map_plate: %s is missing in environment."
+                             "Can't assign map_plate element %s", 
+                             self.value, self.key)
+            if "param" in args_keys:
+                param = self.args["param"].get_value(env)
+                return self.get_param_value(well, param)
+            else:
+                return self.get_well_params(well)
+        else:
+            return self
 
     def get_variable(self, env):
-        pass
+        args_keys = (self.args).keys()
+        if any(i in args_keys for i in ["well", "param"]):
+            var = Variable(self.key, self.get_value(env))
+            return var
+        else:
+            return self
 
-    def get_value(self, env, args = {}):
-        pass
+    def get_wells_ids(self):
+        return list(self.mp_dict.keys())
 
-    def get_param_value(mp_dict, well_name, param):
+    def get_param_value(self, well_name, param):
         """
-        Gets values for a given well and parameter.
+        Gets value for a given well and parameter.
         """
-        value = mp_dict[well_name][param]
+        value = self.mp_dict[well_name][param]
         return value
 
-    def get_param_values(mp_dict, param):
+    def get_param_values(self, param):
         """
         Gets all values for a given param 
         from map_plate dictionary.
         """
         values = []
-        for well in mp_dict:
-            values.append(mp_dict[well][param])
+        for well in self.mp_dict:
+            values.append(self.mp_dict[well][param])
         return values
 
-    def get_param_unique_values(mp_dict, param): 
+    def get_param_unique_values(self, param): 
         """
         Gets all unique values for a given param 
         from map_plate dictionary.
         """
-        values = get_param_values(mp_dict, param)
+        values = get_param_values(self.mp_dict, param)
         unique = list(set(values))
         return unique
 
-    def get_params_names(mp_dict): # get all key names from map_plate dictionary
+    def get_params_names(self): # get all key names from map_plate dictionary
         names = []
-        key_0 = list(mp_dict.keys())[0]
-        names = list(mp_dict[key_0].keys()) 
+        key_0 = list(self.mp_dict.keys())[0]
+        names = list(self.mp_dict[key_0].keys()) 
         #if we assume that number of information between wells is constant
         return names
 
-    def get_well_params(mp_dict, well):
+    def get_well_params(self, well):
         """ 
         Gives all parameters values for a given well
```

Comment:
List. It's inner function. I added comments describing returned value to map_plate functions.